### ISSUE TITLE
CocoaPods and Built in Dynlib Libraries

### ISSUE THREAD

**Issue by NathanaelA** [NativeScript/ios-runtime#380]
Ok, several versions of NativeScript back, #156 was the solution for using Dylib's.   Well, I've attempted to update my nativescript-sqlite for cocoapod support and I can't seem to get it working properly.

Any chance you have someone on staff that has a LOT of experience with cocoapod's that I can pick their brain.   This is totally frustrating as the pod framework file it is building does not seem to have my header or an updated modulemap from the podspec file it is using.

The issue appears to be that for NativeScript to work, I need to reference the spec.library='sqlite3', however; this podspec file does NOT  need to compile anything, since the sqlite3 file is a built in dynamic library on ios.  So basically I need a podspec that tells the xcode project to include another framework library and that is it.   

1. So, I've twisted the podspec file to be able to manage this and the actual compiled file actually looks like it have libsqlite built into the app properly.  (Multiple references inside the compiled emulator app to libsqlite3.dylib).  However, no metadata is being generated.  And my import .h file isn't being copied to the public headers location.   So I have no idea what might be causing the issue.

2. If I manually move my modulemap file into a folder that will get processed by the metagenerator; then the cocoapod works mostly.   By mostly;  The sqlite calls works but throws an error to the console the first time you access a sqlite call:
```
NativeScript could not load library /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk/usr/lib/sqlite3.dylib, error: dlopen(/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk/usr/lib/sqlite3.dylib, 5): image not found
```

To be honest I'm about to just create a simple npm post install script that will automate the install steps I originally used and and documented in #156 to just eliminate the user from having to do anything rather than use cocoapod as cocoapod doesn't seem to be up to the task...

Oh, the podspec is:
```ruby
 Pod::Spec.new do |s|
  s.name     = 'nativescript-sqlite'
  s.version  = '1.0.0'
  s.license  = 'MIT'
  s.summary  = 'SQLite is and embedded SQL database'
  s.authors  = { 'Nathanael Anderson' => 'nathan@master-technology.com' }
  s.requires_arc = false
  s.module_map = 'module.modulemap'
  s.library = 'sqlite3'
  s.platform = :ios
  s.homepage = 'https://github.com/NathanaelA/nativescript-sqlite'
  s.requires_arc = false
  s.source_files = 'platforms/ios/**/*.h'
  s.public_header_files = 'platforms/ios/**/*.h'
  s.source = { :git => 'https://github.com/NathanaelA/nativescript-sqlite.git' }
end
```

Any pointers on how to get this to work without manually moving the modulemap file and how to get this to work so it doesn't throw an error to the console the first time it uses a sqlite call?   Or is this such a edge case that I should just not worry about cocoapod'izing it and just create a npm post install script that adds it to the project file and manually moves the modulemap file.

**Comment by jasssonpet** [comment]
Hello, @NathanaelA,

Regarding this error:
```
NativeScript could not load library sqlite3.dylib, error: dlopen(sqlite3.dylib, 5): image not found
```

it seems that in the newer versions of the iOS SDK, the `sqlite3.dylib` file is renamed to `sqlite3.tbd`.

I found the following quote from Apple:
> For those who are curious, the .tbd files are new "text-based stub libraries", that provide a much more compact version of the stub libraries for use in the SDK, and help to significantly reduce its download size.

It's a bug on our side and we update our symbol loader to search for this file extension as well, so that it can be successfully loaded.

Thank you for reaching us!

**Comment by NathanaelA** [comment]
@jasssonpet -- I'm actually using an older xcode version.  So I don't think that the .tbd is the issue.  The simulator is 8.2.   

Running an older app that I wrote when I first built the sqlite stuff (in NativeScript 1.2) does not throw that error.  It works properly.  I haven't tried to upgrade that code base to NS 1.4 yet; just manually ran the app as it was when I built in in 1.2...

That also doesn't explain why my modulemap isn't being moved/copied/applied in the cocoapod.  Because if I manually stick it somewhere in the metadata generator path (in my case I choose one of the folders shown in the log of where it was looking)

**Comment by NathanaelA** [comment]
Actually I just realized if you look at the error message; it is trying to load "sqlite3.dylib"  The actual file name is "libsqlite3.dylib".   All the cocoapod info on the web say to drop the "lib" prefix on any built in library.  So is NativeScript not automatically re-adding the "lib" prefix when it is trying to load the file?

**Comment by jasssonpet** [comment]
It may very well be so.

**Comment by NathanaelA** [comment]
@jasssonpet, Ok.   I figured out how to fix the error message.  Changing the modulemap file to use 
**link "libsqlite3"** rather than the **link "sqlite3"**
fixes the metadata generation so that when I call a sqlite function nativescript will load the libsqlite3.dylib itself properly.   Not sure if the metageneration tool should be updated to support checking for a "lib*" while building the metadata and then automatically embed the proper name in the metadata.   

However, getting the modulemap installed via a podspec (the main issue) -- I still have no clue.  Any ideas?

**Comment by fealebenpae** [comment]
@NathanaelA You're right, the `lib` prefix should be added by NativeScript when it tries to load the library and we'll fix that.

Now for the modulemap discovery, instead of using a Podfile, you can place a build.xcconfig file in platforms/ios of your plugin along with the modulemap file. You can use the build.xcconfig file to tell Xcode where else to look for modules by extending the `HEADER_SEARCH_PATHS` build setting:
```
HEADER_SEARCH_PATHS = $(inherited) "${PROJECT_DIR}/../../node_modules/nativescript-sqlite/platforms/ios"
```

Initial support for build.xcconfig files in plugins shipped in 1.4, but it is incomplete in that it conflicts with CocoaPods. Full support for build.xcconfig will land in NativeScript for iOS 1.4.1 and CLI 1.4.3, due today.
